components/profile-controller/Makefile

```diff
@@ -0,0 +1,63 @@
+
+# Image URL to use all building/pushing image targets
+IMG ?= gcr.io/kubeflow-images-public/notebook-controller
+TAG ?= $(eval TAG := $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)-$(shell git diff | shasum -a256 | cut -c -6))$(TAG)
+GOLANG_VERSION ?= 1.11.5
+
+all: test manager
+
+# Run tests
+test: generate fmt vet manifests
+	echo "Skip test..."
+	#go test ./pkg/... ./cmd/... -coverprofile cover.out
+
+# Build manager binary
+manager: generate fmt vet
```

Comment:
I tried `make manager` and got
```
$ ☞  make manager
go generate ./pkg/... ./cmd/...
stat ../../vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go: no such file or directory
pkg/apis/apis.go:18: running "go": exit status 1
make: *** [generate] Error 1
```

I also tried `make generate` and `make docker-build` with similar errors.

The problem is deepcopy assumes a vendor directory exists which isn't the case for go modules.
This is due to the fact that kubebuilder doesn't generate a Makefile that works with go modules.

You may want to add the following targets to the Makefile
```
$(GOPATH)/bin/deepcopy-gen:
        GO111MODULE=off $(GO) get k8s.io/code-generator/cmd/deepcopy-gen

deepcopy: $(GOPATH)/bin/deepcopy-gen
        $(GOPATH)/bin/deepcopy-gen -i github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/... -O zz_generated.deepcopy
```

Reply:
will it work if I just checkin vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go?